sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionTests.java

```diff
@@ -0,0 +1,616 @@
+                
+                SELECT sort_array(ARRAY [7, 1, null, 4, null, 3], true);
+                 sort_array
+                ------------
+                 {null, null, 1, 3, 4, 7}
+                (1 row)
+                
+                SELECT sort_array(ARRAY [7e0, 1e0, null, 4e0, null, 3e0], true);
+                 sort_array
+                ------------
+                 {null, null, 1, 3, 4, 7}
+                (1 row)
+                
+                SELECT sort_array(null);
+                 sort_array
+                ------------
+                 NULL
+                (1 row)
+                
+                SELECT sort_array(ARRAY [true, false, null]);
+                 sort_array
+                ------------
+                 {null, false, true}
+                (1 row)
+                
+                SELECT sort_array(ARRAY ['z', 'a', 'c']);
+                 sort_array
+                ------------
+                 { a, c, z}
+                (1 row)
+                
+                SELECT sort_array(ARRAY ['z', 'a', 'c'], false);
+                 sort_array
+                ------------
+                 { z, c, a}
+                (1 row)
+                """
+        );
+    }
+
+    @Test
+    public void testArrayMinMax() {
+        this.qs("""
+                SELECT array_max(ARRAY [9, 1, 2, 4, 8]);
+                 array_max
+                -----------
+                 9
+                (1 row)
+                
+                SELECT array_max(ARRAY [9, 1, 2, 4, 8, null]);
+                 array_max
+                -----------
+                 9
+                (1 row)
+                
+                SELECT array_min(ARRAY [9, 1, 2, 4, 8]);
+                 array_min
+                -----------
+                 1
+                (1 row)
+                
+                SELECT array_min(ARRAY [9, 1, 2, 4, 8, null]);
+                 array_min
+                -----------
+                 1
+                (1 row)
+                """
+        );
+    }
+
+    @Test
+    public void testArrayPosition() {
+        this.qs("""
+                SELECT array_position(ARRAY [2, 4, 6, 8, null], 4);
+                 array_position
+                ----------------
+                 2
+                (1 row)
+                
+                SELECT array_position(ARRAY [2, 4, 6, 8], 4);
+                 array_position
+                ----------------
+                 2
+                (1 row)
+                
+                SELECT array_position(ARRAY [2, 4, 6, 8, null], 3);
+                 array_position
+                ----------------
+                 0
+                (1 row)
+                
+                SELECT array_position(ARRAY [2, 4, 6, 8], 3);
+                 array_position
+                ----------------
+                 0
+                (1 row)
+                
+                SELECT array_position(ARRAY [2, 4, 6, 8], null);
```

Comment:
how about searching for null in an array which contains a null?

Reply:
It looks like that's something that cannot be done... looks like a bug to me.

Reply:
Searching for null elements isn't possible in Spark, and I thought we should stick to it.

Reply:
clearly, but I think it should be emphasized in the documentation